src/server/pfs/cmds/cmds.go

```diff
@@ -663,6 +671,40 @@ want to consider using commit IDs directly.
 			if err != nil {
 				return err
 			}
+
+			// Add trace if env var is set
+			if repo, ok := os.LookupEnv(createPipelineTracingEnvVar); ok && tracing.IsActive() {
+				// unmarshal extended trace from RPC context
+				clientSpan, ctx := opentracing.StartSpanFromContext(
+					c.Ctx(),
+					"/pfs.API/PutFile",
+					ext.SpanKindRPCClient,
+					opentracing.Tag{string(ext.Component), "gRPC"},
+				)
+				defer clientSpan.Finish()
+				extendedTrace := extended.TraceProto{SerializedTrace: map[string]string{}} // init map
+				opentracing.GlobalTracer().Inject(
+					clientSpan.Context(),
+					opentracing.TextMap,
+					opentracing.TextMapCarrier(extendedTrace.SerializedTrace),
+				)
+				extendedTrace.Branch = &pfsclient.Branch{
```

Comment:
There's a `NewBranch` constructor for this, not a huge deal but I figured since this PR cleans up a lot of other raw pb constructors you might as well not add a new one here.

Reply:
:+1: Done